src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -362,14 +363,24 @@ await foreach (var tenant in armClient.GetTenants().GetAllAsync(cancellationToke
             throw new MissingConfigurationException("An azure location/region is required. Set the Azure:Location configuration value.");
         }
 
-        var unique = $"{Environment.MachineName.ToLowerInvariant()}-{environment.ApplicationName.ToLowerInvariant()}";
+        var userSecrets = await userSecretsLazy.Value.ConfigureAwait(false);
+
+        string resourceGroupName;
+        bool createIfAbsent;
 
-        // Name of the resource group to create based on the machine name and application name
-        var (resourceGroupName, createIfAbsent) = _options.ResourceGroup switch
+        if (string.IsNullOrEmpty(_options.ResourceGroup))
         {
-            null or { Length: 0 } => ($"rg-aspire-{unique}", true),
-            string rg => (rg, _options.AllowResourceGroupCreation ?? false)
-        };
+            // Create a unique resource group name and save it in user secrets
+            resourceGroupName = $"rg-aspire-{environment.ApplicationName.ToLowerInvariant()}-{RandomNumberGenerator.GetHexString(8, true)}";
```

Comment:
Lowercase the hex string

Reply:
is even ToLowerInvariant enough? Will RG rule naming allow diacritics, etc? - `azd` likely went through this same exercise already?

Reply:
> Lowercase the hex string

This is what the second argument is for already.

Reply:
While I added diacritics support, I tried and anything seems to work for resource groups:

![image](https://github.com/dotnet/aspire/assets/1165805/dd096d5b-8cb6-4063-8322-29d3671c23a7)

I tried emojis, they passed validation but failed on creation.

Reply:
Just as a note, `azd` doesn't support this as environment name (which gets used in RG creation). Not sure if we want to be consistent but pointing out they use [a-zA-Z0-9] and underscore/dash validation only.

Reply:
In the meantime I added try RG name support.

> this as environment name

You mean "resource group" name, and also that we should actually remove diacritics and then also filter on `[a-zA-Z0-9_\-]` to be sure it works with azd?

Reply:
Is there an overall length constraint as well?

Reply:
> Is there an overall length constraint as well?

Yes. 90 on the RG name, but deployments are limited to 64 and if this gets used with deployment names as well in azureprovisioning, that's something to consider.

Here's what azd does: https://github.com/Azure/azure-dev/blob/6abd371120c98393a8c3ab0c0dba2a0a92285648/cli/azd/pkg/environment/environment.go#L114

Reply:
Now it's normalizing with `azAZ90-_`, it's limiting to 90 chars, and it's also using an optional custom RG prefix because at least for us there is no way to record the ownership of RGs. Tried it locally (i.e. on Azure).